File: ktm-ui/src/app/models/nifty-ltp.ts
export class NiftyLTP {
  id: number;
  niftyTS: string;  // LocalDateTime to Date in TypeScript
  niftyLTP: number;  // Double to number in TypeScript
  meanStrikePCR: number;  // Double to number in TypeScript
  meanRateOI: number;  // Double to number in TypeScript
  combiRate: number;  // Double to number in TypeScript
  support: string;  // String remains string in TypeScript
  resistance: string;  // String remains string in TypeScript
  tradeDecision: string;  // String remains string in TypeScript
  range: string;
  tradeManagement: string;  // String remains string in TypeScript
  tradeHoldCount: number;
  powerTrade: boolean;
  powerTradeType: string;
  iBuyers: boolean;
  iSellers: boolean;
  icebergTrendCounter: number;
  icebergTradeType: string;
  cpts: number;
  maxPainSP: number;  // Double to number in TypeScript
  maxPainSPSecond: number;
  straddleUpside: boolean;
 straddleDownside: boolean;
  closeFull: boolean;
  maxPainCELTP: number;
  maxPainPELTP: number;
 

  constructor() {
    this.id = 0;
    this.niftyTS = '';
    this.niftyLTP = 0;
    this.meanStrikePCR = 0;
    this.meanRateOI = 0;
    this.combiRate = 0;
    this.support = '';
    this.resistance = '';
    this.tradeDecision = '';
    this.range = '';
    this.tradeManagement = '';
    this.tradeHoldCount = 0;
    this.powerTrade = false;
    this.powerTradeType = '';
    this.iBuyers = false;
    this.iSellers = false;
    this.icebergTrendCounter = 0;
    this.icebergTradeType = '';
    this.cpts = 0.0;
    this.maxPainSP = 0;
    this.maxPainSPSecond = 0;
    this.straddleUpside = false;
    this.straddleDownside = false;
    this.closeFull = false;
    this.maxPainCELTP = 0;
    this.maxPainPELTP = 0;

  }
}

File: src/main/java/com/trading/kalyani/KPN/entity/NiftyLTP.java
package com.trading.kalyani.KPN.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Table(name="NiftyLTP")
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class NiftyLTP {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Long id;

    private LocalDateTime niftyTS;

    private Integer niftyLTP;

    private Double meanStrikePCR;

    private Double meanRateOI;

    private Double combiRate;

    private String support;

    private String resistance;

    private String range;

    private String tradeDecision;

    private String tradeManagement;

    private Integer tradeHoldCount;

    private Boolean powerTrade;

    private String powerTradeType;

    private Boolean iBuyers;

    private Boolean iSellers;

    private Integer icebergTrendCounter;

    private String icebergTradeType;

    private Double cpts;

    private Integer maxPainSP;

    private Integer maxPainSPSecond;

    @OneToOne
    @JoinColumn(name = "jobIterationId", referencedColumnName = "id", nullable = false)
    private JobIterationDetails jobIterationDetails;

    @Transient
    private boolean straddleUpside;
    @Transient
    private boolean straddleDownside;
    @Transient
    private boolean closeFull;

    @Transient
    private Double maxPainCELTP;

    @Transient
    private Double maxPainPELTP;

    @Transient
    private Integer lotSize;


}
